microsoft/testsuites/dpdk/dpdkutil.py

```diff
@@ -1085,4 +1099,80 @@ def create_l3fwd_rules_files(
     )
 
 
-DPDK_VERSION_TO_RDMA_CORE_MAP = {"20.11": "46.1", "21.11": ""}
+# Device name match for LSPCI
+class NicType(Enum):
+    CX3 = "ConnectX-3"
+    CX4 = "ConnectX-4"
+    CX5 = "ConnectX-5"
+    MANA = "Device 00ba"
+
+
+# Short name for nic types
+NIC_SHORT_NAMES = {
+    NicType.CX3: "cx3",
+    NicType.CX4: "cx4",
+    NicType.CX5: "cx5",
+    NicType.MANA: "mana",
+}
+
+
+# Get the short name for the type of nic on a node
+def get_node_nic_short_name(node: Node) -> str:
+    devices = node.tools[Lspci].get_devices_by_type(DEVICE_TYPE_SRIOV)
+    if node.nics.is_mana_device_present():
+        return NIC_SHORT_NAMES[NicType.MANA]
+    for nic_name in [NicType.CX3, NicType.CX4, NicType.CX5]:
+        if any([str(nic_name) in x.device_id for x in devices]):
+            return NIC_SHORT_NAMES[nic_name]
+    # We assert much earlier to enforce that SRIOV is enabled,
+    # so we should never hit this unless someone is testing a new platform.
+    # Instead of asserting, just log that the short name was not found.
+    known_nic_types = ",".join(
+        map(str, [NicType.CX3, NicType.CX4, NicType.CX5, NicType.MANA])
+    )
+    found_nic_types = ",".join(map(str, [x.device_id for x in devices]))
+    node.log.debug(
+        "Unknown NIC hardware was detected during DPDK test case. "
+        f"Expected one of: {known_nic_types}. Found {found_nic_types}. "
+    )
+    # this is just a function for annotating a result, so don't assert
+    # if there's
+    return ""
+
+
+# Add dpdk/rdma/nic info to dpdk test result
+# Enables rich reporting, coverage, and issue triage.
+def annotate_dpdk_test_result(
+    test_kit: DpdkTestResources, test_result: TestResult, log: Logger
+) -> None:
+    dpdk_version = None
+    rdma_version = None
+    nic_hw = None
+    try:
+        dpdk_version = test_kit.testpmd.get_dpdk_version()
+    except AssertionError:
+        test_kit.node.log.debug("Could not fetch DPDK version info.")
+    try:
+        rdma_version = test_kit.rdma_core.get_installed_version()
+    except AssertionError:
+        test_kit.node.log.debug("Could not fetch RDMA version info.")
+    try:
+        nic_hw = get_node_nic_short_name(test_kit.node)
+    except AssertionError:
+        test_kit.node.log.debug("Could not fetch NIC short name.")
+    if dpdk_version:
+        test_result.information["dpdk_version"] = str(dpdk_version)
+    if rdma_version:
+        test_result.information["rdma_version"] = str(rdma_version)
+    if nic_hw:
+        test_result.information["nic_hw"] = nic_hw
+
+
+def annotate_dpdk_send_receive_test_result(
+    test_kit: DpdkTestResources, environment: Environment, log: Logger
+) -> None:
+    test_result = environment.source_test_result
```

Comment:
Don't use the `source_test_result`, it may be incorrect, when an env is reused. Use the test result which is passed from test methods.